### PR TITLE
[ESLint] Enable Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Enable Sider's recommended ruleset of 'ESLint'

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

https://github.com/sider/sideci/issues/9964

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

**Review by suetsuguk — COMMENTED** [review]
Please be sure to add a note to CHANGELOG.

File: test/smokes/eslint/yarnrc/package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "dependencies": {
-    "eslint": "7.14.0"
+    "eslint": "7.17.0"
```
[the remark this answers is not in the document]
Comment (sleeeto) [reply test/smokes/eslint/yarnrc/package.json:3]: This older version does not have some rules of our recommended ruleset. And, I updated the version.

File: test/smokes/eslint/yarnrc_yml/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "dependencies": {
-    "eslint": "7.13.0"
+    "eslint": "7.17.0"
```
[the remark this answers is not in the document]
Comment (sleeeto) [reply test/smokes/eslint/yarnrc_yml/package.json:3]: Same.

File: test/smokes/eslint/yarnrc/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (suetsuguk) [reply test/smokes/eslint/yarnrc/package.json:3]: OK. Got it.

File: test/smokes/eslint/yarnrc_yml/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (suetsuguk) [reply test/smokes/eslint/yarnrc_yml/package.json:3]: OK.

**Comment by sleeeto** [comment]
@suetsuguk 
added a note to CHANGELOG.
I want to merge this branch after this week (2021/04/26-4/30) release.

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ Misc:
 - **Metrics Complexity** Add workarounds for lizard's bug [#2249](https://github.com/sider/runners/pull/2249)
 - Show the tool's default version on setup [#2313](https://github.com/sider/runners/pull/2313)
```
Comment (suetsuguk) [inline CHANGELOG.md:23]: [nitpick] remove this line.

**Review by suetsuguk — APPROVED** [review]
LGTM.

**Comment by sleeeto** [comment]
@ybiquitous Ready.

File: CHANGELOG.md [hunk 4]
```diff
@@ -46,6 +46,7 @@ Misc:
 - **RuboCop** Add 3rd-party plugins [#2370](https://github.com/sider/runners/pull/2370)
 - **RuboCop** Fix links [#2371](https://github.com/sider/runners/pull/2371)
 - **CoffeeLint** Drop support for older versions [#2372](https://github.com/sider/runners/pull/2372)
+- **ESLint** Enable new recommended configuration [#2321](https://github.com/sider/runners/pull/2321)
```
Comment (ybiquitous) [inline CHANGELOG.md:49]: Incorrect insertion:

https://github.com/sider/runners/blob/af8781096d379796567896ccb9b5cd57b8992e12/CHANGELOG.md

File: lib/runners/processor/eslint.rb [hunk 5]
```diff
@@ -200,7 +200,7 @@ def run_analyzer(config: nil)
         Results::Success.new(guid: guid, analyzer: analyzer)
       elsif no_eslint_config?(stderr)
         trace_writer.message "Retrying with the default configuration file because no configuration files were found..."
-        FileUtils.copy_file(DEFAULT_ESLINT_CONFIG, ".eslintrc.yml")
+        FileUtils.copy_file(DEFAULT_ESLINT_CONFIG, ".eslintrc.js")
```
Comment (ybiquitous) [inline lib/runners/processor/eslint.rb:203]: [suggestion] Can you refactor it as follows?

```suggestion
        FileUtils.cp(DEFAULT_ESLINT_CONFIG, ".")
```

Comment (sleeeto) [reply lib/runners/processor/eslint.rb:203]: If I changed this, Smoke tests take very long time, and show below errors.

> ? Pattern matching failed at [:result].keys.sort:
--- expected
+++ actual
@@ -1 +1 @@
-[:analyzer, :guid, :issue_count, :issues, :timestamp, :type]
+[:analyzer, :guid, :message, :timestamp, :type]
? Pattern matching failed at [:result][:type]:
Expected: "success"
  Actual: "failure"
? Pattern matching failed at [:result][:message]:
--- expected
+++ actual
@@ -1 +1 @@
-nil
+"The analysis failed due to an unexpected error. See the analysis log for details."
? mismatched_package_version  4821.0s
? Pattern matching failed at [:result].keys.sort:
--- expected
+++ actual
@@ -1 +1 @@
-[:analyzer, :guid, :issue_count, :issues, :timestamp, :type]
+[:analyzer, :guid, :message, :timestamp, :type]
? Pattern matching failed at [:result][:type]:
Expected: "success"
  Actual: "failure"
? Pattern matching failed at [:result][:message]:
--- expected
+++ actual
@@ -1 +1 @@
-nil
+"The analysis failed due to an unexpected error. See the analysis log for details."
? no_ignore  4849.6s

Comment (ybiquitous) [reply lib/runners/processor/eslint.rb:203]: Oh, sorry. My suggestion is wrong. 🙇🏼

**Review by ybiquitous — APPROVED** [review]
Looks good to me. Thanks. 👍🏼 

By the way, when will you delete the old configuration (`sider_eslintrc.yml`)?

**Comment by sleeeto** [comment]
@ybiquitous Removed it.

**Comment by ybiquitous** [comment]
When we merge this PR, the following link ("default ESLint configuration") will be dead. No problems?
https://help.sider.review/tools/javascript/eslint#default-configuration-for-eslint

**Comment by sleeeto** [comment]
@ybiquitous No problem. I am going to update Sider Help also, and There will be not the link of this old ESLint configuration.

**Comment by ybiquitous** [comment]
@sleeeto Are you going to merge sider/sider-docs#607 after merging this PR? Though, we cannot merge the doc PR until we deploy the new Runners version.

**Comment by sleeeto** [comment]
@ybiquitous How about DO NOT delete the old configuration (sider_eslintrc.yml) ?
Does this configuration affects current system ?

**Comment by ybiquitous** [comment]
> Does this configuration affects current system ?

No, we can delete it later. But, I'm worried that we might forget to do it.

**Comment by sleeeto** [comment]
@ybiquitous 
I am going to make a issue that I will remove old configurations until 2021/7/30.
How about this ?

**Comment by ybiquitous** [comment]
Sounds good. 👍🏼 

By the way, where does `2021/7/30` come from?

**Comment by sleeeto** [comment]
@ybiquitous All recommended configurations will be enabled until 6/30.
And I will be ready for removing old configurations until 7/30.

**Comment by ybiquitous** [comment]
Good. Please create an issue with the background resource (blog article?) for `7/30`.

**Comment by sleeeto** [comment]
@ybiquitous I rollbacked the old configuration, and I made a issue for deleting all old configurations.

**Review by ybiquitous — APPROVED** [review]

**Comment by ybiquitous** [comment]
@sleeeto Thank you!

**Pull request merged**